common/help-on-spot-models/src/utils/Database.ts

```diff
@@ -1,41 +1,57 @@
-import { Connection, createConnection } from 'typeorm';
-import {PostgresConnectionOptions} from "typeorm/driver/postgres/PostgresConnectionOptions";
+import { Connection, ConnectionManager, ConnectionOptions, createConnection, getConnectionManager } from 'typeorm'
+import {ormConfig} from "../ormConfig";
 import User from "../entity/User";
-import Address from "../entity/Address";
 import Qualification from "../entity/Qualification";
 import Organisation from "../entity/Organisation";
+import Address from "../entity/Address";
 import Request from "../entity/Request";
 import RequestResponse from "../entity/RequestResponse";
-import {ormConfig} from "../ormConfig";
 import {QualificationsMigration1586981133000} from "../migration/qualificationsMigration";
 
 export class Database {
-  private connection?: Connection;
+  private connectionManager: ConnectionManager
+
+  constructor() {
+    this.connectionManager = getConnectionManager()
+  }
+
+  public async getConnection(): Promise<Connection> {
+    const CONNECTION_NAME = `default`
+
+    let connection: Connection
+
+    if (this.connectionManager.has(CONNECTION_NAME)) {
+      console.info(`Database.getConnection()-using existing connection ...`)
+      connection = await this.connectionManager.get(CONNECTION_NAME)
```

Comment:
do connections need to be refreshed before being reused? 
It was not typeORM but in another project we same really nasty behaviour with outdated connections returning cached results that did not exist anymore, etc

Reply:
How do you "refresh" a connection?